lib/ruby2d/window.rb

```diff
@@ -26,14 +27,19 @@ def initialize(args = {})
       @vsync       = args[:vsync]  || true
       @mouse_x, @mouse_y = 0, 0
       @objects     = []
-      @key, @key_down, @key_held, @key_up = [], [], [], []
-      @mouse, @mouse_down, @mouse_up, @mouse_scroll, @mouse_move = [], [], [], [], []
-      @controller, @controller_axis = [], []
-      @controller_button_down, @controller_button_up = [], []
+      @key, @key_down, @key_held, @key_up = {}, {}, {}, {}
+      @mouse, @mouse_down, @mouse_up, @mouse_scroll, @mouse_move = {}, {}, {}, {}, {}
+      @controller, @controller_axis = {}, {}
+      @controller_button_down, @controller_button_up = {}, {}
+      @event_key = "a"
```

Comment:
If this bothers you, I can change, this does seem like a good number, and introducing code you don't like isn't acceptable, since I'm mostly trying to cover for me, and possibly other people writing lousy code.